itou/antivirus/management/commands/scan_s3_files.py

```diff
@@ -25,11 +26,14 @@ class Command(BaseCommand):
 
     def handle(self, *args, **options):
         now = timezone.now()
-        files = (
-            File.objects.exclude(scan__clamav_completed_at__gt=now - relativedelta(month=1))
-            .order_by("scan__clamav_completed_at")
-            .asc(nulls_first=True)[: self.BATCH_SIZE]
```

Comment:
Tu ne veux pas utiliser `.order_by(models.F("scan__clamav_completed_at").asc(nulls_first=True))` ?

Reply:
https://github.com/betagouv/itou/pull/1853/commits/07b37c1d6f9a08ae1bdc67c1227809665e810a5b

> Actually, the suggestion of using a simple query (without EXISTS) does
not work well with `.select_for_update()`. Attempting to run the
generated query results in:

> django.db.utils.NotSupportedError: FOR NO KEY UPDATE cannot be applied to the nullable side of an outer join

Reply:
J’accepte volontiers de mettre un commentaire :)

Reply:
même avec un `of=('self',)` au `select_for_update` ?

Reply:
Si, bien vu. Je ne connaissais pas de `of=` :)